index.js

```diff
@@ -133,72 +199,105 @@ class CognitoAddCustomAttributesPlugin {
     this.serverless = serverless;
     this.options = options;
     this.provider = serverless.getProvider('aws');
-    
+
     this.hooks = {
-      'after:deploy:deploy': this.postDeploy.bind(this),
+      'after:deploy:deploy': this.postDeploy.bind(this)
     };
-    
+
     this.log = this.log.bind(this);
     this.run = this.run.bind(this);
   }
-  
-  postDeploy() {
-    return new Promise((resolve, reject) => {
-      this.custom = loadCustom(this.log, this.serverless.service.custom);
-      
-      return this.run().then(() => resolve());
-    });
+
+  async postDeploy() {
+    this.custom = loadCustom(this.log, this.serverless.service.custom);
+
+    await this.run();
   }
-  
+
   async run() {
     const { provider: AWS, custom, log } = this;
-    
-    try{
+
+    try {
       log('Start');
-      
+
       const stack = await describeStack(AWS);
-      
-      if(Array.isArray(custom)) {
-        custom.forEach(async (mappingItem) => {
-          // skip empty mapping items
-          if(_.isEmpty(mappingItem)) {
-            return;
-          }
-          
-          // if we run more than one at a time, there is a chance for a race condition on creating
-          // the customAttribute on the userPool the first time
-          await this.processCognitoCustomAttributeMapping(stack, mappingItem);
-        });
+
+      if (Array.isArray(custom)) {
+        await Promise.all(
+          custom.map((mappingItem) =>
+            (async () => {
+              // skip empty mapping items
+              if (_.isEmpty(mappingItem)) {
+                return;
+              }
+
+              // if we run more than one at a time, there is a chance for a race condition on creating
+              // the customAttribute on the userPool the first time
+              await this.processCognitoCustomAttributeMapping(
+                stack,
+                mappingItem
+              );
+            })()
+          )
+        );
```

Comment:
This is the main fix, replacing the `forEach` for a `map` returning a promise (the async function is invoked inline) and wrapping it with `Promise.all`.

Reply:
LGTM